adhocteam/Head-Start-TTADP
# TTAHUB-193: Clear Condition on Topic Change

## Description of change

We need to make sure that if the filter topic changes that we also clear the condition. The reason is that if they user is switching between date and string filters the previous date filters are NOT valid for strings (BEFORE, IN, AFTER).

## How to test

Create a report filter for 'Start Date' with the 'before' condition and any date then apply. 

Now switch the topic of the filter to 'Creator'. The condition field should now be cleared as 'before' is not valid for a string field like creator. 


## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-193


## Checklists

### Every PR


~- [ ] API Documentation updated~
~- [ ] Boundary diagram updated~
~- [ ] Logical Data Model updated~
~- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions~

### Production Deploy

~- [ ] Staging smoke test completed~

### After merge/deploy
